fendermint/module/src/genesis.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+//! Genesis module trait for initializing module-specific actors.
+//!
+//! This trait allows modules to participate in genesis state creation
+//! by initializing their own actors and state.
+
+use anyhow::Result;
+use cid::Cid;
+use fendermint_vm_genesis::Genesis;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_shared::address::Address;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::ActorID;
+
+/// State context provided to genesis modules.
+///
+/// This provides access to the state tree and other genesis parameters
+/// that modules need to initialize their actors.
+///
+/// # Note on Generic Methods
+///
+/// This trait is generic over some type parameters, making it not directly
+/// trait-object-safe. Implementations should use concrete types when
+/// calling these methods.
```

Comment:
Maybe consider making obj-safe by introducing a generic type parameter `S: serde::Serialize` instead of using `impl serde::Serialize` in the `create_custom_actor` method?